app/src/main/java/org/oppia/android/app/spotlight/SpotlightFragment.kt

```diff
@@ -0,0 +1,430 @@
+import org.oppia.android.databinding.BottomLeftOverlayBinding
+import org.oppia.android.databinding.BottomRightOverlayBinding
+import org.oppia.android.databinding.TopLeftOverlayBinding
+import org.oppia.android.databinding.TopRightOverlayBinding
+import org.oppia.android.domain.spotlight.SpotlightStateController
+import org.oppia.android.util.accessibility.AccessibilityService
+import org.oppia.android.util.data.AsyncResult
+import org.oppia.android.util.data.DataProviders.Companion.toLiveData
+import java.util.LinkedList
+import javax.inject.Inject
+
+/**
+ * Fragment to hold the spotlights on elements. This fragments provides a single place for all the
+ * spotlight interactions, and handles lifecycle related functionality for spotlights, such as
+ * surviving orientation changes and marking spotlights as seen.
+ */
+class SpotlightFragment : InjectableFragment(), SpotlightNavigationListener, SpotlightManager {
+  @Inject
+  lateinit var activity: AppCompatActivity
+
+  @Inject
+  lateinit var spotlightStateController: SpotlightStateController
+
+  @Inject
+  lateinit var accessibilityService: AccessibilityService
+
+  @Inject
+  lateinit var resourceHandler: AppLanguageResourceHandler
+
+  private var targetList = LinkedList<Target>()
+  private lateinit var spotlight: Spotlight
+  private var screenHeight: Int = 0
+  private var screenWidth: Int = 0
+  private lateinit var overlayBinding: Any
+  private var internalProfileId: Int = -1
+  private var isSpotlightActive = false
+  private val isRtl by lazy {
+    resourceHandler.getLayoutDirection() == ViewCompat.LAYOUT_DIRECTION_RTL
+  }
+
+  private fun calculateScreenSize() {
+    val displayMetrics = DisplayMetrics()
+    activity.windowManager.defaultDisplay.getMetrics(displayMetrics)
+
+    screenHeight = displayMetrics.heightPixels
+    screenWidth = displayMetrics.widthPixels
+  }
+
+  // since this fragment does not have any view to inflate yet, all the tasks should be done here.
+  override fun onAttach(context: Context) {
+    super.onAttach(context)
+    (fragmentComponent as FragmentComponentImpl).inject(this)
+
+    internalProfileId = arguments?.getInt(PROFILE_ID_ARGUMENT_KEY) ?: -1
+    calculateScreenSize()
+  }
+
+  /**
+   * Requests a spotlight to be shown on the [SpotlightTarget]. The spotlight is enqueued if it
+   * [SpotlightTarget] hasn't been shown before in a FIFO buffer. This API can ensure proper
+   * spotlighting of a which is laid out after an animation. For the spotlights to work correctly,
+   * we must know the [SpotlightTarget]'s anchor's size and position. For a view that is laid out
+   * after an animation, we must wait until the final size and positions of the anchor view are
+   * measured, which can be achieved by using doOnPreDraw. Refer:
+   * (https://betterprogramming.pub/stop-using-post-postdelayed-in-your-android-views-9d1c8eeaadf2)
+   *
+   * @param spotlightTarget The [SpotlightTarget] for which the spotlight is requested
```

Comment:
For this & the next KDoc, since the interface versions of these are already present, we can remove these ones. Instead, please move implementation-specific details into line comments within the methods and keep the SpotlightManager versions of these methods' KDocs specific to what the method does (rather than how it does it).

Reply:
done